controllers/helpers.go

```diff
@@ -158,88 +158,102 @@ func GetCloudProviderSecret(d azure.ClusterDescriber, namespace, name string, ow
-	return json.MarshalIndent(config, "", "    ")
+func userAssignedIdentityCloudProviderConfig(d azure.ClusterDescriber, identityID string) (*CloudProviderConfig, *CloudProviderConfig) {
+	controlPlaneConfig, workerConfig := newCloudProviderConfig(d)
+	controlPlaneConfig.AadClientID = ""
+	controlPlaneConfig.AadClientSecret = ""
+	controlPlaneConfig.UseManagedIdentityExtension = true
+	controlPlaneConfig.UserAssignedIdentityID = identityID
+	return controlPlaneConfig, workerConfig
 }
 
-func userAssignedIdentityCloudProviderConfig(d azure.ClusterDescriber, identityID string) ([]byte, error) {
-	config := newCloudProviderConfig(d)
-	config.AadClientID = ""
-	config.AadClientSecret = ""
-	config.UseManagedIdentityExtension = true
-	config.UserAssignedIdentityID = identityID
-	return json.MarshalIndent(config, "", "    ")
-}
-
-func newCloudProviderConfig(d azure.ClusterDescriber) *CloudProviderConfig {
+func newCloudProviderConfig(d azure.ClusterDescriber) (controlPlaneConfig *CloudProviderConfig, workerConfig *CloudProviderConfig) {
 	return &CloudProviderConfig{
-		Cloud:                        d.CloudEnvironment(),
-		AadClientID:                  d.ClientID(),
-		AadClientSecret:              d.ClientSecret(),
-		TenantID:                     d.TenantID(),
-		SubscriptionID:               d.SubscriptionID(),
-		ResourceGroup:                d.ResourceGroup(),
-		SecurityGroupName:            d.NodeSubnet().SecurityGroup.Name,
-		SecurityGroupResourceGroup:   d.ResourceGroup(),
-		Location:                     d.Location(),
-		VMType:                       "vmss",
-		VnetName:                     d.Vnet().Name,
-		VnetResourceGroup:            d.Vnet().ResourceGroup,
-		SubnetName:                   d.NodeSubnet().Name,
-		RouteTableName:               fmt.Sprintf("%s-node-routetable", d.ClusterName()),
-		LoadBalancerSku:              "Standard",
-		MaximumLoadBalancerRuleCount: 250,
-		UseManagedIdentityExtension:  false,
-		UseInstanceMetadata:          true,
-	}
+			Cloud:                        d.CloudEnvironment(),
+			AadClientID:                  d.ClientID(),
+			AadClientSecret:              d.ClientSecret(),
+			TenantID:                     d.TenantID(),
+			SubscriptionID:               d.SubscriptionID(),
+			ResourceGroup:                d.ResourceGroup(),
+			SecurityGroupName:            d.NodeSubnet().SecurityGroup.Name,
+			SecurityGroupResourceGroup:   d.ResourceGroup(),
+			Location:                     d.Location(),
+			VMType:                       "vmss",
+			VnetName:                     d.Vnet().Name,
+			VnetResourceGroup:            d.Vnet().ResourceGroup,
+			SubnetName:                   d.NodeSubnet().Name,
+			RouteTableName:               fmt.Sprintf("%s-node-routetable", d.ClusterName()),
+			LoadBalancerSku:              "Standard",
+			MaximumLoadBalancerRuleCount: 250,
+			UseManagedIdentityExtension:  false,
+			UseInstanceMetadata:          true,
+		},
+		&CloudProviderConfig{
```

Comment:
we can probably reduce the worker config a lot further, but fine for a first iteration while we work out the credentials since they're most important